### PR TITLE
Add Reliability to DBotScore Class

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Description
Added the reliability argument to the DBotScore class to support the Source reliability option in XSOAR.
relates: https://github.com/demisto/etc/issues/32672

@EliorKedar FYI

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by anara123 — CHANGES_REQUESTED** [review]
add unit test

**Review by anara123 — APPROVED** [review]

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -2119,6 +2119,9 @@ class DBotScore(object):
         :type malicious_description: ``str``
         :param malicious_description: if the indicator is malicious and have explanation for it then set it to this field
 
+        :type reliability: ``str``
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2122]: This should be an enum. There are a fixed set of allowed values.

**Review by glicht — CHANGES_REQUESTED** [review]

**Comment by glicht** [comment]
> add unit test

Please also add a playbook test. It is important we have a good test end-to-end.

File: Packs/Base/TestPlaybooks/playbook-Test-CreateDBotScore-With-Reliability.yml [hunk 2]
```diff
@@ -0,0 +1,790 @@
...
+      scriptName: CreateDBotScoreEntry
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "7"
+    scriptarguments:
+      indicator_type:
+        simple: ip
```
Comment (yaakovi) [inline Packs/Base/TestPlaybooks/playbook-Test-CreateDBotScore-With-Reliability.yml:55]: should be domain

File: Packs/Base/TestPlaybooks/playbook-Test-CreateDBotScore-With-Reliability.yml [hunk 3]
```diff
@@ -0,0 +1,790 @@
...
+      scriptName: CreateDBotScoreEntry
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "23"
+    scriptarguments:
+      indicator_type:
+        simple: ip
```
Comment (yaakovi) [inline Packs/Base/TestPlaybooks/playbook-Test-CreateDBotScore-With-Reliability.yml:580]: should be domain

**Review by yaakovi — APPROVED** [review]
looks good to me

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 4]
```diff
@@ -247,6 +247,41 @@ def is_valid_type(cls, _type):
...
 
+class DBotScoreReliability(object):
+    """
+    Enum: Source reliability levels
+    Values are case sensitive
+
+    :return: None
+    :rtype: ``None``
+    """
+    THIRD_PT_ENRICHMENT = 'A+ - 3rd party enrichment'
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:258]: Can we change the naming of these constants to: A_PLUS, A, B, ...

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 5]
```diff
@@ -247,6 +247,41 @@ def is_valid_type(cls, _type):
...
+    FAIRLY = 'C - Fairly reliable'
+    NOT_USUALLY = 'D - Not usually reliable'
+    UNRELIABLE = 'E - Unreliable'
+    CANNOT_BE_JUDGED = 'F - Reliability cannot be judged'
+
+    def __init__(self):
+        # required to create __init__ for create_server_docs.py purpose
+        pass
+
+    @classmethod
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:270]: Small remark: It is better practice to use staticmethod in such cases.

**Review by glicht — DISMISSED** [review]

**Review by yaakovi — APPROVED** [review]

**Pull request merged**